tools/cpp/windows_cc_configure.bzl

```diff
@@ -227,9 +228,38 @@ def find_msvc_tool(repository_ctx, vc_path, tool):
     tool_path = vc_path + "\\bin\\amd64\\" + tool
 
   if not repository_ctx.path(tool_path).exists:
-    auto_configure_fail(tool_path + " not found, please check your VC++ installation.")
+    return None
+
   return tool_path
 
+def _check_vc_installation(repository_ctx, vc_path):
+  """Check if the VC++ installation is valid."""
+  if vc_path == "visual-studio-not-found":
+    tpl(repository_ctx, "vc_path_not_found.bat", {"%{vc_error_message}" : ""})
```

Comment:
Creating the "vc_path_not_found.bat" is a side-effect of this function, and I wouldn't expect this side effect when looking at the call site and the name of the function. Could you remove the side effect (the template expansion) and do it at the call site? This method should (`_check_vc_installation`) really just check things.